packages/blade/src/components/Carousel/Carousel.native.tsx

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable consistent-return */
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable react/jsx-no-useless-fragment */
+import React from 'react';
+import { ScrollView } from 'react-native';
+import { Indicators } from './Indicators/Indicators';
+import { NavigationButton } from './NavigationButton';
+import type { CarouselProps } from './types';
+import type { CarouselContextProps } from './CarouselContext';
+import { CarouselContext } from './CarouselContext';
+import BaseBox from '~components/Box/BaseBox';
+import { useInterval } from '~utils/useInterval';
+import { makeAccessible } from '~utils/makeAccessible/makeAccessible.native';
+import { announce } from '~components/LiveAnnouncer/LiveAnnouncer.native';
+import { castNativeType } from '~utils';
+
+const percentageStringToNumber = (percentage: string) => {
+  if (!percentage.endsWith('%')) {
```

Comment:
No this is native specific and we need to do this to calculate the snap points in RN. 

on react-native we have to accept a percentage, i've updated the types based on this too. 

Anyways on mobile it doesn't make much sense to put carouselitemWidth="500px", it will either be 100% (full width) or <100% (bleed if needed)